traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -1006,67 +1075,132 @@ func updateV40(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, dsV40 *
 		}
 	}
 
-	resultRows, err := tx.Query(updateDSQuery(),
-		&ds.Active,
-		&ds.CCRDNSTTL,
-		&ds.CDNID,
-		&ds.CheckPath,
-		&deepCachingType,
-		&ds.DisplayName,
-		&ds.DNSBypassCNAME,
-		&ds.DNSBypassIP,
-		&ds.DNSBypassIP6,
-		&ds.DNSBypassTTL,
-		&ds.DSCP,
-		&ds.EdgeHeaderRewrite,
-		&ds.GeoLimitRedirectURL,
-		&ds.GeoLimit,
-		&ds.GeoLimitCountries,
-		&ds.GeoProvider,
-		&ds.GlobalMaxMBPS,
-		&ds.GlobalMaxTPS,
-		&ds.FQPacingRate,
-		&ds.HTTPBypassFQDN,
-		&ds.InfoURL,
-		&ds.InitialDispersion,
-		&ds.IPV6RoutingEnabled,
-		&ds.LogsEnabled,
-		&ds.LongDesc,
-		&ds.LongDesc1,
-		&ds.LongDesc2,
-		&ds.MaxDNSAnswers,
-		&ds.MidHeaderRewrite,
-		&ds.MissLat,
-		&ds.MissLong,
-		&ds.MultiSiteOrigin,
-		&ds.OriginShield,
-		&ds.ProfileID,
-		&ds.Protocol,
-		&ds.QStringIgnore,
-		&ds.RangeRequestHandling,
-		&ds.RegexRemap,
-		&ds.RegionalGeoBlocking,
-		&ds.RemapText,
-		&ds.RoutingName,
-		&ds.SigningAlgorithm,
-		&ds.SSLKeyVersion,
-		&ds.TenantID,
-		&ds.TRRequestHeaders,
-		&ds.TRResponseHeaders,
-		&ds.TypeID,
-		&ds.XMLID,
-		&ds.AnonymousBlockingEnabled,
-		&ds.ConsistentHashRegex,
-		&ds.MaxOriginConnections,
-		&ds.EcsEnabled,
-		&ds.RangeSliceBlockSize,
-		&ds.Topology,
-		&ds.FirstHeaderRewrite,
-		&ds.InnerHeaderRewrite,
-		&ds.LastHeaderRewrite,
-		&ds.ServiceCategory,
-		&ds.MaxRequestHeaderBytes,
-		&ds.ID)
+	if omitExtraLongDescFields {
+		if ds.LongDesc1 != nil || ds.LongDesc2 != nil {
+			return nil, http.StatusBadRequest, errors.New("the Long Description 1 and Long Description 2 fields are no longer supported in API 4.0 onwards"), nil
```

Comment:
this error message should probably use the json field names, i.e. `longDesc1` and `longDesc2` just to be clear